src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/Animations/AnimationTypes/TadaAnimationType.shared.cs

```diff
@@ -34,33 +34,64 @@ public double MinimumScale
 		}
 
 		public static readonly BindableProperty RotationAngleProperty =
-		   BindableProperty.Create(nameof(RotationAngle), typeof(double), typeof(TadaAnimationType), TadaAnimation.DefaultRotationAngle, BindingMode.TwoWay);
+		   BindableProperty.Create(nameof(RotationAngle), typeof(double), typeof(TadaAnimationType), 3.0, BindingMode.TwoWay);
 
 		public double RotationAngle
 		{
 			get => (double)GetValue(RotationAngleProperty);
 			set => SetValue(RotationAngleProperty, value);
 		}
 
-		protected override uint DefaultDuration { get; set; } = TadaAnimation.DefaultLength;
+		protected override uint DefaultDuration { get; set; } = 1000;
 
 		public override Task Animate(View? view)
 		{
-			if (view != null)
-			{
-				var taskCompletionSource = new TaskCompletionSource<bool>();
+			if (view == null)
+				return Task.FromResult(false);
+
+			var taskCompletionSource = new TaskCompletionSource<bool>();
+
+			CreateAnimation(
+				16,
+				onFinished: (v, c) =>
+				{
+					if (IsRepeated)
+					{
+						return;
+					}
+
+					taskCompletionSource.SetResult(c);
+				},
+				shouldRepeat: () => IsRepeated,
+				view).Commit();
+
+			return taskCompletionSource.Task;
+		}
+
+		protected override Animation CreateAnimation(params View[] views) => Create(RotationAngle, MinimumScale, MaximumScale, views);
```

Comment:
Here we have a situation. If you do something like

```csharp
		public Test()
		{
			var tada = new TadaAnimationType();
			tada.IsRepeated = true;
			tada.MaximumScale = 1;
			var lbl = new Label();
			var entry = new Entry();

			// This works with more than one view
			var animation = tada.CreateAnimation(16, null, null, lbl, entry);
			animation.Commit();
			animation.Abort();

			// This just works with one view
			tada.Animate(lbl); // Task
		}
```

This can lead to confusion about which method should I call? Just `Animate` or `CreateAnimation`?
Also, if I call `Animate` I can't `Abort` it.

Reply:
So, if we want to use the BehaviorAnimation API, we need to make those animations simpler and drop the `CreateAnimation`, `Commit`, and `Abort` methods, which works for me.

If you see that those methods, at least  `Commit`, and `Abort`, we can :

1. Try to add those methods in BehaviorAnimation API;
2. Create another API, to hold those animations.

Just to point out, that these animations are more complex than the animations that we have, so makes sense to be another implementation if the existing one doesn't fit well.

Reply:
I completely agree on the confusion this creates. What do you think to keeping the BehaviorAnimation support but creating a second underlying type that BehaviorAnimation can use?

Currently we have AnimationBase, could this stay but then we create 2 abstract classes inheriting from this; SimpleAnimation and ComplexAnimation (or possibly AsyncAnimation and Animation). Then the BehaviorAnimation class could call them based on which is supplied.

Or an alternative could be to bend the complex to work in the existing approach by allowing a CancellationToken to be passed in for the aborting. This all feels a bit wrong considering non of the animations are actually async.

Reply:
I guess there are good use cases for both async and non async still. I was going to suggest we could replace the async parts with implementing these directly in an on async manner: https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Core/ViewExtensions.cs but I am not sure that is a good idea now